### ISSUE TITLE
set-has-size disappeared?

### ISSUE THREAD

**Issue by LeventErkok** [Z3Prover/z3#3854]
For this benchmark:

```
(set-logic ALL)
(declare-fun s0 () (Array Int Bool))
(declare-fun s1 () Int)
(define-fun s2 () Bool (set-has-size s0 s1))
```

a fresh build of z3 is saying:

```
(error "line 4 column 42: unknown function/constant set-has-size")
```

This used to work just fine up until about a week ago. (Maybe even later.)

Was the `set-has-size` function removed recently?

**Comment by rainoftime** [comment]
@LeventErkok  Yes. It is removed from the SMT-Lib2 frontend. Perhaps you can use the C++/Python API.

**Comment by LeventErkok** [comment]
@rainoftime I'm maintaining an upstream tool that communicates with z3 via SMTLib. So, C++/Python isn't really an option.

Having said that, while it would be nice to have it back, it's not that particularly important. I can remove it from my own API similarly.

I'm curious though why remove it only from SMTLib interface if you're still supporting it through other API?

**Comment by rainoftime** [comment]
@LeventErkok 
Here is the commit https://github.com/Z3Prover/z3/commit/5516e420a138fe9b696cb47815ee3f7994892261 disabling bapa SMTlib interface.
It should quite easy to "re-open" the features in your local git branch (if possible)

**Comment by agurfinkel** [comment]
@rainoftime , what is the reason for disabling `bapa` in smt-lib interface. Is that only to prevent the code from being fuzzed improperly? Seems strange if this creates problems to users.

**Comment by rainoftime** [comment]
@muchang @wintered @5hadowblad3
Perhaps z3 can bring `bapa` SMT-lib2 interface back, and we do not generally set constraints with the two operations (`set-has-size` and `card`)?

@NikolajBjorner

**Comment by NikolajBjorner** [comment]
there are too many fuzz bugs on small stuff. BAPA is not a mainstream feature so I don't want to support it if it has one bug after the other. Mainstream features can be supported

**Comment by LeventErkok** [comment]
@NikolajBjorner Should I interpret this as `set-has-size` and `card` are never coming back to the SMTLib interface, or that they'll be enabled again once the fuzz work is complete?

**Comment by LeventErkok** [comment]
@NikolajBjorner 

Based on comments at #5672, I'm concluding that `set-has-size` and `card` are not going to be introduced anytime soon. I'm closing this ticket.

**Comment by NikolajBjorner** [comment]
I don't know about your use case, but with sufficient set of use cases it makes sense to support features. 
, they need to be done well or not at all.
